pkg/logger/worker.go

```diff
@@ -81,6 +87,29 @@ func (w *Worker) sendCloudEvent(logReq LogRequest) error {
 		return fmt.Errorf("while creating http transport: %s", err)
 	}
 
+	if logReq.Url.Scheme == "https" {
+		caCertFilePath := filepath.Join(LoggerCaCertMountPath, logReq.CertName)
+		caCertFile, err := os.ReadFile(caCertFilePath)
+		// Do not fail if certificates not found, for backwards compatibility
+		if err == nil {
+			clientCertPool := x509.NewCertPool()
+			if !clientCertPool.AppendCertsFromPEM(caCertFile) {
+				return fmt.Errorf("while parsing CA certificate")
+			}
+
+			tlsTransport := &http.Transport{
+				TLSClientConfig: &tls.Config{
+					RootCAs:            clientCertPool,
+					MinVersion:         tls.VersionTLS12,
+					InsecureSkipVerify: logReq.TlsSkipVerify, // #nosec G402
+				},
+			}
+			t.Client.Transport = tlsTransport
+		} else {
+			w.Log.Warnf("using https endpoint but could not find CA cert file %s", caCertFilePath)
```

Comment:
wouldn't it fail to send the events if the target is using SSL but there is no client certificate?